src/HalResource.php

```diff
@@ -384,12 +368,6 @@ private function aggregateEmbeddedCollection(string $name, array $collection, st
             ));
         }
 
-        $this->compareResources(
-            $originalFirstResource,
-            $collectionFirstResource,
-            $name,
-            $context
-        );
         return $original + $collection;
```

Comment:
Unrelated, is it intentionally using `+` here?
Numeric keys would be used from `$original` and values for same keys from `$collection` would be discarded

Reply:
Not sure: I kept it as-is, but indeed, `+` is not my preference for arrays.